docs/source/bots.rst

```diff
@@ -0,0 +1,114 @@
+###############
+Orchestra Bots
+###############
+
+Below, we'll walk through the various bots available in Orchestra.
+
+*********
+StaffBot
+*********
+
+``StaffBot`` provides a simple way to ask a group of workers if they would like
+to work on a particular ``Task``. The goal is reduce the human load when
+finding a ``Worker`` for  a ``Task``. ``StaffBot`` allows a user to interact
+with Orchestra via `Slack`_ or can be configured to automatically find workers for
+a ``Task`` when it is available by setting up an `Assignment Policy`_.
+
+``StaffBot`` works by reaching out to qualified workers and offering them a
+``Task`` to work on. Workers can then either accept or reject the task and
+start working on it.
+
+
+Slack
+=====
+
+To interact with ``StaffBot`` via Slack, you first need to configure a `Slack
+Slash Command <https://api.slack.com/slash-commands>`_. Below is a sample
+configuration for the ``/staffbot`` command.
+
+
+.. image:: ../static/img/bots/slash_command_config_token.png
+.. image:: ../static/img/bots/slash_command_autocomplete.png
+
+Once the command is created, copy the token for the command into your project
+settings as follows: ``ORCHESTRA_SLACK_STAFFBOT_TOKEN = 'your-token-here'``.
+The token is used to authenticated requests sent to the staffbot url. It is
+important to keep this token secret since otherwise anyone may make a ``HTTP
+POST`` and execute the staffing logic!
+
+In addition to the ``ORCHESTRA_SLACK_STAFFBOT_TOKEN`` setting, you can further
+restrict access to the endpoint by specifying the following::
+
+ STAFFBOT_CONFIG = {
+            'allowed_team_ids': ['allowed_team_ids'],
+            'allowed_team_domains': ['allowed_domains'],
+            'allowed_channel_ids': ['allowed_channel_ids'],
+            'allowed_channel_names': ['allowed_channel_names'],
+            'allowed_user_ids': ['allowed_user_ids'],
+            'allowed_user_names': ['allowed_user_names'],
+            'allowed_commands': ['allowed_commands'],
+        }
+
+``StaffBot`` will use the ``STAFFBOT_CONFIG`` to filter messages that do not
+match items in the list. If a parameter is not specified, all values are
+accepted by default.
+
+
+Once this configuration is complete, you can test by typing ``/staffbot staff
+<task-id>`` where ``<task-id>`` is an unassigned task id.
```

Comment:
"By default, StaffBot will reach out to ``Workers`` who have the appropriate ``WorkerCertification`` for the task you are staffing, in batches of X ``Workers`` every Y minutes.